### DISCUSSION TITLE
How can we handle Application throttling in PnP core SDK while connecting SharePoint online

### DISCUSSION THREAD

**Discussion by arpan2512** [pnp/pnpcore#1268] [Q&A]
We are getting "A Task was canceled" Error while connecting SharePoint online using PnP Core SDK in windows console application (C#), we have updated the below configuration in appsettings.json file. But still we are getting the error.  Could you please suggest how to handle this issue with example.

"PnPCore": {
    "DisableTelemetry": "false",
    "HttpRequests": {
      "UserAgent": "ISV|Company|ProductX",
      "Timeout": "100",
      "RateLimiterMinimumCapacityLeft": "20",
      "SharePointRest": {
        "UseRetryAfterHeader": "true",
        "MaxRetries": "10",
        "DelayInSeconds": "10",
        "UseIncrementalDelay": "true"
      },
      "MicrosoftGraph": {
        "UseRetryAfterHeader": "true",
        "MaxRetries": "10",
        "DelayInSeconds": "10",
        "UseIncrementalDelay": "true"
      }
    },

**Comment by arpan2512** [comment]
How to implement EventHub (retryEvent) in PnP Core SDK? any sample code?

**Reply by jansenbe** [reply]
@arpan2512 : can you check the returned error, not sure the "A task was cancelled" error is network request related? Doing a Fiddler trace will clarify this or capturing the exception details.

About using EventHub, we've integrated that in the Microsoft 365 Assessment tool, here's a link to get you started: https://github.com/pnp/pnpassessment/blob/b1c4a61433f2768cb2d81a1a037d9dcfcd66fd78/src/PnP.Scanning/PnP.Scanning.Core/Services/ScanManager.cs#L560-L598
